tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs

```diff
@@ -89,18 +120,98 @@ public class Interpreter : IDisposable
           tensor, tensorDataPtr, Buffer.ByteLength(outputTensorData)));
     }
 
+    public TensorInfo GetInputTensorInfo(int index) {
+      TfLiteTensor tensor = TfLiteInterpreterGetInputTensor(interpreter, index);
+      return GetTensorInfo(tensor);
+    }
+
+    public TensorInfo GetOutputTensorInfo(int index) {
+      TfLiteTensor tensor = TfLiteInterpreterGetOutputTensor(interpreter, index);
+      return GetTensorInfo(tensor);
+    }
+
+    /// <summary>
+    /// Returns a string describing version information of the TensorFlow Lite library.
+    /// TensorFlow Lite uses semantic versioning.
+    /// </summary>
+    /// <returns>A string describing version information</returns>
+    public static string GetVersion() {
+      return Marshal.PtrToStringAnsi(TfLiteVersion());
+    }
+
+    private static string GetTensorName(TfLiteTensor tensor) {
+      return Marshal.PtrToStringAnsi(TfLiteTensorName(tensor));
+    }
+
+    private static TensorInfo GetTensorInfo(TfLiteTensor tensor) {
+      int[] dimensions = new int[TfLiteTensorNumDims(tensor)];
+      for (int i = 0; i < dimensions.Length; i++) {
+        dimensions[i] = TfLiteTensorDim(tensor, i);
+      }
+      return new TensorInfo() {
+        name = GetTensorName(tensor),
+        type = TfLiteTensorType(tensor),
+        dimensions = dimensions,
+        quantizationParams = TfLiteTensorQuantizationParams(tensor),
+      };
+    }
+
     private static void ThrowIfError(int resultCode) {
       if (resultCode != 0) throw new Exception("TensorFlowLite operation failed.");
     }
 
     #region Externs
 
+    #if UNITY_IPHONE && !UNITY_EDITOR
+    private const string TensorFlowLibrary = "__Internal";
+#else
+    private const string TensorFlowLibrary = "tensorflowlite_c";
+#endif
+
+    public enum TfLiteType {
```

Comment:
I'd be OK calling this class `DataType`, and TfLiteQuantizationParams just `QuantizationParams`. That way we nicely distinguish between the native C, opaque TfLite* types.